src/python/TaskWorker/Actions/PostJob.py

```diff
@@ -956,6 +956,13 @@ def updateOrInsertDoc(self, doc, toTransfer):
                 msg += " Transfer submission failed."
                 msg += "\n%s" % (str(hte.headers))
                 returnMsg['error'] = msg
+            # add/remove some keys from newDoc to has the same schema as `if not self.found_doc_in_db` is True.
```

Comment:
why doing it here and not in the definition of newDoc a few lines above ?

Reply:
You are right.

Add `type` back should be fine and nothing bad for have extra key in transfers.txt.
I need to test to make sure REST is working properly.

Reply:
Unfortunately, I need to add extra validation for that https://github.com/dmwm/CRABServer/compare/master...novicecpp:CRABServer:add_type_back_to_transfer_dicts_prcompare?expand=1

In my opinion, `newDoc` should construct before if-else, then you copy the object and manipulate it as you like for talking to REST. But it is a big change. So, I am still leaning toward my original PR here.

What do you think?

Reply:
I am too tired to understand. Can we have a short chat tomorrow ?

Reply:
Sure.